### PR TITLE
Add github build & test actions, disable travis.

### PR DESCRIPTION
**Pull request by squeed**
RIP travis.

Fixes: #554

### CODE REVIEW

**Comment by squeed** [comment]
Okay, this is good to go. @dcbw

**Comment by squeed** [comment]
Okay, after a lot of wrangling, I got windows working too. And fixed a racy test.

This is ready to merge.

File: build_linux.sh [hunk 1]
```diff
@@ -25,9 +15,9 @@ PLUGINS="plugins/meta/* plugins/main/* plugins/ipam/*"
 for d in $PLUGINS; do
 	if [ -d "$d" ]; then
 		plugin="$(basename "$d")"
-		if [ $plugin != "windows" ]; then
+		if [ "${plugin}" != "windows" ]; then
 			echo "  $plugin"
-			$GO build -o "${PWD}/bin/$plugin" "$@" "$REPO_PATH"/$d
+      ${GO:-go} build -o "${PWD}/bin/$plugin" "$@" ./"$d"
```
Comment (dcbw) [inline build_linux.sh:20]: @squeed whitespace?

**Comment by dcbw** [comment]
lgtm except for the whitespace thing

**Comment by dcbw** [comment]
LGTM

**Pull request merged**